packages/next/src/utils.ts

```diff
@@ -2445,6 +2445,12 @@ export async function getMiddlewareBundle({
           shortPath.replace(/^app\//, '').replace(/(^|\/)page$/, '') || 'index';
       }
 
+      if (routesManifest?.basePath) {
+        shortPath = path.posix
+          .join(routesManifest.basePath, shortPath)
+          .replace(/^\//, '');
```

Comment:
I'm curious about this `.replace(/^\//, '')` because I'm not sure how its related to basePath.

Is this a different bug you found?

Reply:
It appears the `routesManifest.basePath` contains the unaltered value from the `next.config.js` and thus a leading slash is valid. The build system is quite happy with or without the leading slash, however every key in the `edgeFunctions` object is a path that does _not_ start with a leading slash, so for consistency's sake I remove the leading slash and things still work as they were.

I could have simply stripped the leading slash from `routesManifest.basePath` before passing it into `join()`, but I felt this looks easier to read.